pkg/download/download.go

```diff
@@ -82,29 +83,32 @@ func Download(endpoint string, opts ...Option) (b []byte, err error) {
 			return retry.ExpectedError(err)
 		}
 
-		if dlOpts.Format == b64 {
-			var b64 []byte
-
-			b64, err = base64.StdEncoding.DecodeString(string(b))
-			if err != nil {
-				return err
-			}
-
-			b = b64
-		}
-
 		return nil
 	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to download config from %q: %w", u.String(), err)
 	}
 
+	if dlOpts.Format == b64 {
+		var b64 []byte
+
+		b64, err = base64.StdEncoding.DecodeString(string(b))
+		if err != nil {
+			return nil, err
+		}
+
+		b = b64
+	}
+
 	return b, nil
 }
 
-// download handles the actual http request
 func download(req *http.Request) (data []byte, err error) {
-	client := &http.Client{}
+	transport := &http.Transport{}
```

Comment:
I really hate that in Go's stdlib, but empty structure is a bad initializer as http's DefaultTransport is not empty: https://godoc.org/net/http#RoundTripper

Most important is missing `Proxy` function iirc, which allows to parse proxy from the env.

One option is to copy directly by value and use that as initial value:

`transport := *(http.DefaultTransport.(*http.Transport))`

Reply:
not sure if I got the syntax correctly, but I hope the idea is clear

Reply:
Good point! Fixed.